### ISSUE TITLE
[bug] expose `conan.tools.apple.apple_deployment_target_flag` (#12872)

### ISSUE THREAD

**Issue by SSE4** [conan-io/conan#13029]
### Environment details

* Operating System+version: `Linux debian 6.1.8-x64v3-custom-xanmod1 #1 SMP PREEMPT_DYNAMIC Tue Jan 31 19:37:53 +07 2023 x86_64 GNU/Linux`
* Compiler+version: `g++ (GCC) 12.2.0`
* Conan version: `Conan version 2.0.0-beta9`
* Python version: `Python 3.9.2`

### Steps to reproduce

1. `python -c "from conan.tools.apple import apple_deployment_target_flag"`

### Logs

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: cannot import name 'apple_deployment_target_flag' from 'conan.tools.apple' (/home/sse4/.local/lib/python3.9/site-packages/conan/tools/apple/__init__.py)
```

**Comment by paulharris** [comment]
Required for PR https://github.com/conan-io/conan-center-index/pull/15353

**Comment by SpaceIm** [comment]
duplicate of https://github.com/conan-io/conan/issues/12604

In conan v2, the equivalent of `tools.apple.apple_deployment_target_flag()` is a private function `conan.tools.apple.apple.apple_min_version_flag()`. As a workaround you can indirectly get a return of this function with `AutotoolsToolchain(self).apple_min_version_flag` like `boost` recipe does.

**Comment by SSE4** [comment]
`apple_min_version_flag` hits zero search results in conan docs for me.
if it's undocumented, it's probably a subject to change, this doesn't sound like a reliable workaround.

**Comment by SpaceIm** [comment]
Sure, I said it was a private function, and https://github.com/conan-io/conan/issues/12604 asks to make it public. `AutotoolsToolchain(self).apple_min_version_flag` aims to provide the result of this logic (workaround as I said), so I don't understand your answer.

**Comment by SSE4** [comment]
I mean both `conan.tools.apple.apple_min_version_flag` and `AutotoolsToolchain(self).apple_min_version_flag` are private/undocummented (while, it seems like `AutotoolsToolchain(self).apple_arch_flag` and `AutotoolsToolchain(self).apple_isysroot_flag` are documented, it's a bit inconsistent).

**Comment by lieser** [comment]
Needed for the Botan recipe as well (https://github.com/conan-io/conan-center-index/pull/18079). Going with the `AutotoolsToolchain(self).apple_min_version_flag` workaround for now.

**Comment by memsharded** [comment]
Closing as duplicated in https://github.com/conan-io/conan/issues/12604
